velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +342,43 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
+ public:
+  explicit LikeConstantPattern(StringView pattern, char escapeChar)
+      : re_(toStringPiece(likePatternToRe2(pattern, escapeChar)), RE2::Quiet) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      Expr* /* caller */,
+      EvalCtx* context,
+      VectorPtr* resultRef) const final {
+    VELOX_CHECK(args.size() == 2 || args.size() == 3);
+
+    // apply() will not be invoked if the selection is empty.
+    checkForBadPattern(re_);
+
+    exec::LocalDecodedVector toSearch(context, *args[0], rows);
```

Comment:
This function has 3 arguments, two of which are constant. Therefore, the first non-constant argument will be either flat or constant.

Reply:
I was following the logic of the other functions in this file. Are you implying that there should be separate cases for a flat  and constant vector instead ?

Reply:
Since input can only be either flat or constant, it could be more efficient to handle these two cases explicitly and not rely on DecodedVector and virtual function calls involved.